matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/MediaScanRestClient.java

```diff
@@ -167,7 +170,35 @@ public void onSuccess(String serverPublicKey) {
                 }
 
                 if (request != null) {
-                    request.enqueue(new DefaultRetrofit2CallbackWrapper<>(callback));
+                    request.enqueue(new DefaultRetrofit2CallbackWrapper<>(new SimpleApiCallback<MediaScanResult>(callback) {
+                        @Override
+                        public void onSuccess(MediaScanResult scanResult) {
+                            callback.onSuccess(scanResult);
+                        }
+
+                        @Override
+                        public void onNetworkError(Exception exception) {
+                            // Check whether the provided encrypted_body could not be decrypted.
+                            if (exception != null && exception instanceof HttpException) {
+                                HttpError error = ((HttpException) exception).getHttpError();
+                                if (error.getHttpCode() == 403) {
+                                    MediaScanError mcsError;
+                                    try {
+                                        String bodyAsString = error.getErrorBody();
+                                        mcsError = JsonUtils.getGson(false).fromJson(bodyAsString, MediaScanError.class);
+                                    } catch (Exception e) {
+                                        mcsError = null;
+                                    }
+                                    if (mcsError != null && mcsError.reason == MediaScanError.MCS_BAD_DECRYPTION) {
```

Comment:
In Java we cannot use `==` to compare String. It's ok in kotlin.
Please replace by: "MediaScanError.MCS_BAD_DECRYPTION.equals(mcsError.reason)"